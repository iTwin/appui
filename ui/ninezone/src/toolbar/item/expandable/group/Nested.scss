@import "./_variables";

.nz-toolbar-item-expandable-group-nested {
  .nz-back-arrow-container {
    width: 15px;
    height: 20px;
    position: absolute;
    left: 6px;
    top: 8px;

    &:hover {
      cursor: pointer;

      > * {
        color: $primary-tint;
      }
    }

    > * {
      position: relative;
      left: 3px;
      top: 3px;
    }
  }
}
