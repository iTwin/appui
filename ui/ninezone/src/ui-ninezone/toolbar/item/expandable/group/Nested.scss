/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/

@import "_variables";

.nz-toolbar-item-expandable-group-nested {
  position: relative;

  .nz-back-arrow-container {
    width: 15px;
    height: 20px;
    position: absolute;
    left: 6px;
    top: 8px;

    &:hover {
      cursor: pointer;

      > * {
        color: $primary-tint;
      }
    }

    > * {
      position: relative;
      left: 3px;
      top: 3px;
    }
  }
}
