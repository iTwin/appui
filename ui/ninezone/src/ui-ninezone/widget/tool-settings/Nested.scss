/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "_variables";

.nz-widget-toolSettings-nested {
  display: block;

  > .nz-header {
    display: grid;
    align-items: center;
    justify-items: center;
    grid-template-columns: auto 1fr;
    grid-column-gap: $nz-tool-settings-content-padding;
    padding-bottom: $nz-tool-settings-content-padding;

    > .nz-button {
      grid-row: 1;
    }

    > .nz-title {
      grid-row: 1;
    }
  }
}
