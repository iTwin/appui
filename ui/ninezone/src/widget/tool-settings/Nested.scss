/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
.nz-widget-toolSettings-nested {
  display: block;

  > .nz-header {
    display: grid;
    align-items: center;
    justify-items: center;
    grid-template-columns: auto 1fr;
    grid-column-gap: 5px;
    margin-bottom: 5px;

    > .nz-button {
      grid-row: 1;
      width: 25px;
      height: 25px;
      font-size: 16px;
    }

    > .nz-label {
      grid-row: 1;
    }
  }
}
