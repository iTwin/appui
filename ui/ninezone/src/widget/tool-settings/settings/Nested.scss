.nz-widget-toolSettings-settings-nested {
  display: block;

  > .nz-header {
    display: grid;
    align-items: center;
    justify-items: center;
    grid-auto-columns: auto 1fr;
    grid-column-gap: 5px;
    margin-bottom: 5px;

    > .nz-button {
      grid-row: 1;
      width: 25px;
      height: 25px;

      > .nz-back-icon {
        > * {
          width: 15px;
          height: 15px;
        }
      }
    }

    > .nz-label {
      grid-row: 1;
    }
  }
}
