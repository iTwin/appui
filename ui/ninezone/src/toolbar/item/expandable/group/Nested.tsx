/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module Toolbar */

import * as classnames from "classnames";
import * as React from "react";
import { withContainIn, WithContainInProps } from "../../../../base/WithContainIn";
import { Panel } from "./Panel";
import { GroupProps } from "./Group";
import { Columns } from "./Columns";
import { Title } from "./Title";
import { BackArrow } from "./BackArrow";
import "./Nested.scss";

/** Properties of [[NestedGroup]] component. */
export interface NestedGroupProps extends GroupProps {
  /** Function called when the back arrow is clicked. */
  onBack?: () => void;
}

/** Nested tool group component. Used in [[ExpandableItem]] component. */
export class NestedGroup extends React.PureComponent<NestedGroupProps> {
  public render() {
    const className = classnames(
      "nz-toolbar-item-expandable-group-nested",
      this.props.className);

    return (
      <Panel className={className} style={this.props.style}>
        <div
          className="nz-back-arrow-container"
          onClick={this.props.onBack}
        >
          <BackArrow />
        </div>
        <Title>
          {this.props.title}
        </Title>
        <Columns>
          {this.props.columns}
        </Columns>
      </Panel>
    );
  }
}

// tslint:disable-next-line:variable-name
export const NestedWithContainIn: React.ComponentClass<NestedGroupProps & WithContainInProps> = withContainIn(NestedGroup);
