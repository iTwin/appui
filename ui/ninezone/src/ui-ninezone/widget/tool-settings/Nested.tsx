/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module ToolSettings */

import * as classnames from "classnames";
import * as React from "react";
import { ToolSettings, ToolSettingsProps } from "./Settings";
import "./Nested.scss";

/** Properties of [[Nested]] component. */
export interface NestedProps extends ToolSettingsProps {
  /** Nested settings label. */
  label?: string;
  /** Back button icon. */
  backButton?: React.ReactNode;
}

/** Nested tool settings component. Used as content of [[ToolSettings]]. */
export class Nested extends React.PureComponent<NestedProps> {
  public render() {
    const className = classnames(
      "nz-widget-toolSettings-nested",
      this.props.className);

    return (
      <ToolSettings
        className={className}
        style={this.props.style}
      >
        <div className="nz-header">
          <div className="nz-button">
            {this.props.backButton}
          </div>
          <div className="nz-label">
            {this.props.label}
          </div>
        </div>
        {this.props.children}
      </ToolSettings>
    );
  }
}
