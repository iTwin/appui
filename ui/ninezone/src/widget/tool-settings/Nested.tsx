/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module ToolSettings */

import * as classnames from "classnames";
import * as React from "react";
import Settings, { SettingsProps } from "./Settings";
import "./Nested.scss";

/** Properties of [[Nested]] component. */
export interface NestedProps extends SettingsProps {
  /** Nested settings label. */
  label?: string;
  /** Back button icon. */
  backButton?: React.ReactNode;
}

/** Nested tool settings component. Used as content of [[ToolSettings]]. */
export default class Nested extends React.Component<NestedProps> {
  public render() {
    const className = classnames(
      "nz-widget-toolSettings-nested",
      this.props.className);

    return (
      <Settings
        className={className}
        style={this.props.style}
      >
        <div className="nz-header">
          <div className="nz-button">
            {this.props.backButton}
          </div>
          <div className="nz-label">
            {this.props.label}
          </div>
        </div>
        {this.props.children}
      </Settings>
    );
  }
}
