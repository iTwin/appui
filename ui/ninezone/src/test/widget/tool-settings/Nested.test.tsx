/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
import { mount, shallow } from "enzyme";
import * as React from "react";

import Nested from "../../..//widget/tool-settings/Nested";

describe("<Nested />", () => {
  it("should render", () => {
    mount(<Nested />);
  });

  it("renders correctly", () => {
    shallow(<Nested />).should.matchSnapshot();
  });
});
