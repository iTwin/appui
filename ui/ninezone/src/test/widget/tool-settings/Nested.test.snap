// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nested /> renders correctly 1`] = `
<ToolSettings
  className="nz-widget-toolSettings-nested"
>
  <div
    className="nz-header"
  >
    <div
      className="nz-button"
    />
    <div
      className="nz-label"
    />
  </div>
</ToolSettings>
`;
